### PR TITLE
Soc 2499

### PR DESCRIPTION
**Pull request by qaga**
- new discussions architecture
- moved methods from PostsList and PostDetails classes

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public void waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public void waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/OptionsPostAndReply.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/OptionsPostAndReply.java]: MINOR Remove this unused import 'java.util.List'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* CRITICAL 2 critical
* MAJOR 3 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/63340f6b2c8099bf52e7545e431577182ef8aca7/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [2 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/63340f6b2c8099bf52e7545e431577182ef8aca7/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/63340f6b2c8099bf52e7545e431577182ef8aca7/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/OptionsPostAndReply.java (hunk 3 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/OptionsPostAndReply.java]: MINOR Remove this unused import 'java.util.List'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* CRITICAL 2 critical
* MAJOR 3 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/7a069a360188006874a62b4eb2186ce6ecacae01/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [2 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/7a069a360188006874a62b4eb2186ce6ecacae01/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/7a069a360188006874a62b4eb2186ce6ecacae01/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* CRITICAL 2 critical
* MAJOR 3 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/e0f1cf41ca44c23233fea303d90c0773cb1ee6b4/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [2 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/e0f1cf41ca44c23233fea303d90c0773cb1ee6b4/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/e0f1cf41ca44c23233fea303d90c0773cb1ee6b4/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* CRITICAL 2 critical
* MAJOR 3 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/584a58d32b732eb49274fc5908671f966028cbf3/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [2 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/584a58d32b732eb49274fc5908671f966028cbf3/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/584a58d32b732eb49274fc5908671f966028cbf3/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostDetailsPage.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.wikia.webdriver.elements.mercury.pages.discussions;
+
+import com.wikia.webdriver.elements.mercury.components.discussions.common.Post;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.Reply;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import lombok.Getter;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostDetailsPage.java]: MINOR Remove this unused import 'org.openqa.selenium.support.FindBy'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostDetailsPage.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.List;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostDetailsPage.java]: MINOR Remove this unused import 'java.util.List'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 2 major
* MINOR 4 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/457af9da51dbae5bf4ac2c7964f0a34c4595a538/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L44) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/457af9da51dbae5bf4ac2c7964f0a34c4595a538/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L51) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

File: src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java [hunk 6]
```diff
@@ -9,12 +9,16 @@
 import com.wikia.webdriver.common.core.helpers.User;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.elements.mercury.components.Loading;
-import com.wikia.webdriver.pageobjectsfactory.pageobject.discussions.PostsListPage;
+import com.wikia.webdriver.elements.mercury.components.discussions.desktop.SortingTool;
+import com.wikia.webdriver.elements.mercury.components.discussions.mobile.DiscussionsHeader;
+import com.wikia.webdriver.elements.mercury.components.discussions.mobile.FiltersPopOver;
+import com.wikia.webdriver.elements.mercury.components.discussions.mobile.FiltersPopOver;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java]: MINOR Remove this duplicated import. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/OptionsPostAndReply.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+/**
+ * @ownership Social Wikia
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/OptionsPostAndReply.java]: remove ownership

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+/**
+ * @ownership Social Wikia
+ */
+public class Post extends WikiBasePageObject{
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: Extend just BasePageObject

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    return this;
+  }
+
+  public Post clickUsernameLink() {
+    avatarUsername.click();
+    return this;
+  }
+
+  public Post scrollToBottom(WebDriver driver) {
+    ((JavascriptExecutor) driver)
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: no need to cast to JavascripExecutor

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java [hunk 10]
```diff
@@ -0,0 +1,144 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: Change all void returns to Post returns

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/UserHeroUnit.java [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+/**
+ * @ownership Social Wikia
+ */
+public class UserHeroUnit extends WikiBasePageObject{
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/UserHeroUnit.java]: extend just a BasePageObject

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/CommunityBadge.java [hunk 12]
```diff
@@ -0,0 +1,10 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.desktop;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+/**
+ * @ownership Social Wikia
+ */
+public class CommunityBadge extends WikiBasePageObject{
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/CommunityBadge.java]: Extend Just BasePageObject

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/ContributorsList.java [hunk 13]
```diff
@@ -0,0 +1,10 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.desktop;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+/**
+ * @ownership Social Wikia
+ */
+public class ContributorsList extends WikiBasePageObject{
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/ContributorsList.java]: Extend just BasePageObject

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/HeroUnit.java [hunk 14]
```diff
@@ -0,0 +1,10 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.desktop;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+/**
+ * @ownership Social Wikia
+ */
+public class HeroUnit extends WikiBasePageObject{
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/HeroUnit.java]: Extend Just BasePageObject

File: src/test/java/com/wikia/webdriver/testcases/discussions/NavigatingTests.java [hunk 15]
```diff
@@ -9,12 +9,14 @@
 import com.wikia.webdriver.common.core.drivers.Browser;
 import com.wikia.webdriver.common.core.helpers.User;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
-import com.wikia.webdriver.pageobjectsfactory.pageobject.discussions.PostsListPage;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.Post;
+import com.wikia.webdriver.elements.mercury.components.discussions.desktop.BackButtons;
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/testcases/discussions/NavigatingTests.java]: Add an enter before assertion

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public void waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 2 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/e24d46ee766d48672e4df5d82069cd447649cf62/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L44) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/e24d46ee766d48672e4df5d82069cd447649cf62/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L51) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

File: src/test/java/com/wikia/webdriver/testcases/discussions/NavigatingTests.java (hunk 15 above)
Comment (qaga) [inline src/test/java/com/wikia/webdriver/testcases/discussions/NavigatingTests.java]: Added spaces before assertions, I believe this comment was made in wrong line? ;)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java (hunk 10 above)
Comment (qaga) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: done

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 2 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/f3d3bc6485038e92db247f12e489244f861116e7/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L44) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/f3d3bc6485038e92db247f12e489244f861116e7/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L51) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java (hunk 9 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: moved method to jsActions and reused inherited jsActions from BasePageObject

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java (hunk 16 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: added logError

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public Post waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      PageObjectLogging.logError("waitForVoteCountChangeTimeLagToPass", e);
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public void waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      PageObjectLogging.logError("waitForVoteCountChangeTimeLagToPass", e);
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Reply.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 2 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/6a81e6dd398d3d3e77e18ae268d55fdc73d0948d/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L44) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/6a81e6dd398d3d3e77e18ae268d55fdc73d0948d/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L51) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 2 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/81ad1919b6c4e3669ce3411292471539b9f83e87/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L51) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/81ad1919b6c4e3669ce3411292471539b9f83e87/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L44) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

**Comment by justnpT** [comment]
Both WikiaTech = sonarqube comments are incorrect. The error message is logged.
@ludwikkazmierczak can we do something about WikiaTech being incorrect ?

**Comment by justnpT** [comment]
I fixed the remaining CR notes and checked that all tests passed: http://jenkins:8080/view/Social/job/Test-Case-Automation-Discussions/66/

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 2 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/11136ef7c71191cd13153da39f5abfb85abd1374/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L51) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/11136ef7c71191cd13153da39f5abfb85abd1374/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L44) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/8bc09bcf79070cbc86b2c102b3b892ae044af8b9/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L51) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
* MAJOR [Move this closing curly brace to the next line.](https://github.com/Wikia/selenium-tests/blob/8bc09bcf79070cbc86b2c102b3b892ae044af8b9/src/test/java/com/wikia/webdriver/testcases/discussions/SharingTests.java#L44) [rule](http://sonar-s1/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

**Comment by ludwikkazmierczak** [comment]
LGTM

**Pull request merged**